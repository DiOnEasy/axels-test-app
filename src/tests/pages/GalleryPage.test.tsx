import '@babel/preset-react';
import '@testing-library/jest-dom';
import { screen, waitFor } from '@testing-library/react';

import configureStore from 'redux-mock-store';
import { renderWithRouter } from 'tests/helpers/renderWithRouter';

const mockStore = configureStore([]);

describe('GalleryPage', () => {
    let store: any;
    const initialState = {
        images: {
            images: [
                { id: '1', url: 'image1.jpg' },
                { id: '2', url: 'image2.jpg' }
            ],
            loading: true
        }
    };

    beforeEach(() => {
        store = mockStore(initialState);
    });

    const getStateWithLoading = (loading: boolean) => ({
        ...initialState,
        images: {
            ...initialState.images,
            loading
        }
    });

    it('should render loading state initially', () => {
        renderWithRouter({ route: '/gallery' }, store);

        expect(screen.getByText('loading')).toBeInTheDocument();
    });

    it('should render images after loading', async () => {
        store = mockStore(getStateWithLoading(false));

        renderWithRouter({ route: '/gallery' }, store);

        await waitFor(() => {
            expect(screen.getAllByRole('img')).toHaveLength(2);
        });
    });

    it('should match a snapshot', () => {
        store = mockStore(getStateWithLoading(false));

        const { baseElement } = renderWithRouter(
            {
                route: '/gallery'
            },
            store
        );

        expect(baseElement).toMatchSnapshot();
    });
});
