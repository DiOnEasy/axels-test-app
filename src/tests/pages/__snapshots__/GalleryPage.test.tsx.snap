// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage GalleryPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-y8o7sp-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1oljlpb-MuiGrid-root"
    >
      <a
        href="/gallery/1"
      >
        <img
          class="sc-eqUzNf dtNceb"
          src="image1.jpg"
        />
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1oljlpb-MuiGrid-root"
    >
      <a
        href="/gallery/2"
      >
        <img
          class="sc-eqUzNf dtNceb"
          src="image2.jpg"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
